modules/widgets/src/zoom-widget.tsx

```diff
@@ -56,11 +56,14 @@ export class ZoomWidget implements Widget<ZoomWidgetProps> {
     this.viewId = props.viewId || null;
     this.placement = props.placement || 'top-left';
     this.orientation = props.orientation || 'vertical';
-    props.transitionDuration = props.transitionDuration || 200;
-    props.zoomInLabel = props.zoomInLabel || 'Zoom In';
-    props.zoomOutLabel = props.zoomOutLabel || 'Zoom Out';
-    props.style = props.style || {};
-    this.props = props;
+
+    this.props = {
+      ...props,
+      transitionDuration: props.transitionDuration || 200,
```

Comment:
Wonder if it would be nice to support `defaultProps` for widgets to avoid this sort of code

Reply:
`Widget` is only an interface, so I think the best we can do is lead by example of least annoying code to maintain. I just changed the dev guide to use `this.id = props.id ?? this.id` to avoid needing to repeat the hard-coded values, which is a marked improvement over this.

Reply:
Could it be an abstract class instead?

Reply:
It's tempting but I'm more in favor of flexibility at this stage. I'm hesitant to abstract too early across widgets with there are so many ways to implement UI. 

It's a bit of extra work to fill out the interface, but not much.

Maybe when we have a fuller library of examples exploring a breadth of use cases.. until then, there's nothing preventing an author from making their own abstract class.